src/Surfnet/StepupSelfService/SelfServiceBundle/Service/RemoteVetting/Encryption/IdentityEncrypter.php

```diff
@@ -40,18 +41,75 @@ public function __construct(RemoteVettingConfiguration $configuration, IdentityW
         $this->writer = $writer;
     }
 
-    /**
-     * @param AttributeListDto $identity
-     */
     public function encrypt(AttributeListDto $identity)
     {
         $data = $identity->serialize();
-        $publicKey = $this->configuration->getPublicKey();
+        $rsaPublicKey = $this->configuration->getPublicKey();
+
+        if (!is_string($data) || !is_string($rsaPublicKey)) {
+            // Invalid argument
+            throw new InvalidArgumentException('Invalid input was provided to the encrypt method');
+        }
+
+        // Use AES-256 in GCM
+        $symmetricAlgorithm = 'aes-256-gcm';
+
+        // Generate initialisation vector for the symmetric encryption algorithm
+        $ivLength = openssl_cipher_iv_length($symmetricAlgorithm);
+        if (false === $ivLength) {
+            // Error generating key
+            throw new InvalidArgumentException(
+                'Unable to generate an initialization vector (iv) based on the selected symmetric encryption algorithm'
+            );
+        }
+
+        $iv = openssl_random_pseudo_bytes($ivLength);
+        if (false === $iv) {
+            // Error generating key
+            throw new InvalidArgumentException('Unable to generate a correct initialization vector (iv)');
+        }
+
+        // Generate a 256 bits AES key
+        $secretKey = openssl_random_pseudo_bytes(256 / 8);
+        if (false === $secretKey) {
+            // Error generating key
+            throw new InvalidArgumentException('Unable to generate the secret key');
+        }
+
+        // Encrypt the data
+        $tag = '';
+        $ciphertext = openssl_encrypt($data, $symmetricAlgorithm, $secretKey, 0, $iv, $tag);
```

Comment:
Could we still support 5.6 for R17 switching. Maybe with the help of a polyfill or PKCS15?
Consider creating an issue so we could drop 5.6 in the end?

Reply:
I've looked for a polyfill, but have not found one specifically suited for backporting the new openssl_encrypt methods (guess nobody's willing to burn their fingers on this)

We could implement the PKCS15 method, use a development only service replacement for the encrypter, or simply accept the fact we should use PHP72 in this development effort.

Reply:
Couldn't we just use the FakeIdentityWriter in dev and skip the IdentityEncrypter tests for PHP5.6 for the time being?

Reply:
Yes, sounds like a plan. I'll add this as a task/chore in the backlog